pkg/controllers/pod_controller_test.go

```diff
@@ -0,0 +1,430 @@
+			jobName:           "test-jobset-replicated-job-1-test-job-0",
+			podName:           "test-jobset-replicated-job-1-test-job-0-1",
+			ns:                ns,
+			jobIdx:            0})
+		wantPodWrapper = makePod(&makePodArgs{
+			jobSetName:        jobSetName,
+			replicatedJobName: "replicated-job-1",
+			jobName:           "test-jobset-replicated-job-1-test-job-0",
+			podName:           "test-jobset-replicated-job-1-test-job-0-1",
+			ns:                ns,
+			jobIdx:            0})
+	)
+	tests := []struct {
+		name            string
+		pods            []corev1.Pod
+		wantPodsDeleted []corev1.Pod
+		wantErr         error
+		forceClientErr  bool
+	}{
+		{
+			name: "delete follower pods",
+			pods: []corev1.Pod{
+				leaderPodWrapper.AddAnnotation(batchv1.JobCompletionIndexAnnotation, "0").Obj(),
+				followerPodWrapper.Obj(),
+			},
+			wantPodsDeleted: []corev1.Pod{
+				wantPodWrapper.ResourceVersion("999").SetConditions([]corev1.PodCondition{{
+					Type:               corev1.DisruptionTarget,
+					Status:             corev1.ConditionTrue,
+					Reason:             constants.ExclusivePlacementViolationReason,
+					LastTransitionTime: metav1.Now(),
+					Message:            constants.ExclusivePlacementViolationMessage,
+				},
+				}).Obj(),
+			},
+		},
+		{
+			name: "delete follower pods with pod conditions status is false",
+			pods: []corev1.Pod{
+				leaderPodWrapper.AddAnnotation(batchv1.JobCompletionIndexAnnotation, "0").Obj(),
+				followerPodWrapper.SetConditions([]corev1.PodCondition{{
+					Type:               corev1.DisruptionTarget,
+					Status:             corev1.ConditionFalse,
+					Reason:             constants.ExclusivePlacementViolationReason,
+					LastTransitionTime: metav1.Now(),
+					Message:            constants.ExclusivePlacementViolationMessage,
+				},
+				}).Obj(),
+			},
+			wantPodsDeleted: []corev1.Pod{
+				wantPodWrapper.ResourceVersion("999").SetConditions([]corev1.PodCondition{{
+					Type:               corev1.DisruptionTarget,
+					Status:             corev1.ConditionTrue,
+					Reason:             constants.ExclusivePlacementViolationReason,
+					LastTransitionTime: metav1.Now(),
+					Message:            constants.ExclusivePlacementViolationMessage,
+				},
+				}).Obj(),
+			},
+		},
+		{
+			name: "delete follower pods with update pod status error",
+			pods: []corev1.Pod{
+				leaderPodWrapper.AddAnnotation(batchv1.JobCompletionIndexAnnotation, "0").Obj(),
+				followerPodWrapper.SetConditions([]corev1.PodCondition{{
+					Type:               corev1.DisruptionTarget,
+					Status:             corev1.ConditionFalse,
+					Reason:             constants.ExclusivePlacementViolationReason,
+					LastTransitionTime: metav1.Now().Rfc3339Copy(),
+					Message:            constants.ExclusivePlacementViolationMessage,
+				},
+				}).Obj(),
+			},
+			wantPodsDeleted: nil,
+			forceClientErr:  true,
+			wantErr:         errors.Join(errors.Join(errors.New("example error"))),
```

Comment:
```suggestion
			wantErr:         errors.New("example error"),
```

Same for other places with this

Reply:
I previously used `errors.New` for handling errors, but I encountered the following error that indicated a type mismatch.
<img width="1219" alt="image" src="https://github.com/kubernetes-sigs/jobset/assets/86391540/5075bd9f-74a0-49b9-b4da-3211ae2e6c09">

Reply:
I see, what about `fmt.Errorf`, will that work or does it still require a join?

Reply:
use `fmt.Errorf`the same error
<img width="1149" alt="image" src="https://github.com/kubernetes-sigs/jobset/assets/86391540/5d739613-83b3-4aff-8a6a-dda2fc19434b">

Reply:
ok sounds good